src/platforms/macos/daemon/macosdaemonserver.cpp

```diff
@@ -41,6 +41,8 @@ int MacOSDaemonServer::run(QStringList& tokens) {
 
   MacOSDaemon daemon;
 
+  daemon.maybeRecover();
```

Comment:
I don't love the name `maybeRecover`. You're doing a `cleanUpIfNecessary`. Not that that's much better. `maybeCleanup`? It's just not really a recovery so much as a reset or fresh start.